File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    List<Integer> x = new ArrayList<>();
    public List<Integer> rightSideView(TreeNode root) {
        adder(root,0);
        return x;
        
    }
    public void adder(TreeNode node, int level)
    {
        if(node == null)
        {
            return;
        }

        if(x.size() == level)
        {
            x.add(node.val);
        }
        adder(node.right, level+1);
        adder(node.left,level+1);
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []

        def helper(node, d):
            if not node:
                return None
            if len(res) == d:
                res.append([])

            res[d].append(node.val)
            helper(node.left, d+1)
            helper(node.right, d+1)
        helper(root,0)
        return res
        